store/postgres/asset_repository.go

```diff
@@ -649,6 +634,46 @@ func (r *AssetRepository) getAssetVersionSQL() sq.SelectBuilder {
 		LeftJoin("users u ON a.updated_by = u.id")
 }
 
+func (r *AssetRepository) buildFilterQuery(builder sq.SelectBuilder, cfg asset.Config) sq.SelectBuilder {
+	if len(cfg.Types) > 0 {
+		builder = builder.Where(sq.Eq{"type": cfg.Types})
+	}
+
+	if len(cfg.Services) > 0 {
+		builder = builder.Where(sq.Eq{"service": cfg.Services})
+	}
+
+	if len(cfg.QueryFields) > 0 && cfg.Query != "" {
+		orClause := sq.Or{}
+		for _, field := range cfg.QueryFields {
+			orClause = append(orClause, sq.ILike{
+				field: fmt.Sprint("%", cfg.Query, "%"),
+			})
+		}
+		builder = builder.Where(orClause)
+	}
+
+	if len(cfg.Data) > 0 {
+		for key, val := range cfg.Data {
+			builder = builder.Where(fmt.Sprintf("data ->> '%s' = '%s'", key, val))
+		}
+	}
+
+	return builder
+}
+
+func (r *AssetRepository) buildOrderQuery(builder sq.SelectBuilder, cfg asset.Config) sq.SelectBuilder {
+	if cfg.SortBy != "" {
+		orderDirection := "ASC"
+		if cfg.SortDirection != "" {
+			orderDirection = strings.ToUpper(cfg.SortDirection)
+		}
+		return builder.OrderBy(cfg.SortBy + " " + orderDirection)
+	}
```

Comment:
Let's try to return early to avoid nested indentation.

```suggestion
        if cfg.SortBy == "" {
                return builder
        }
        
        orderDirection := "ASC"
	if cfg.SortDirection != "" {
	        orderDirection = cfg.SortDirection
	}
			
	return builder.OrderBy(cfg.SortBy + " " + orderDirection)
```